pangeo_forge_recipes/dynamic_target_chunks.py

```diff
@@ -0,0 +1,140 @@
+import itertools
+from typing import Dict, List, Union
+
+import numpy as np
+import xarray as xr
+from dask.utils import parse_bytes
+
+from pangeo_forge_recipes.aggregation import XarraySchema, schema_to_template_ds
+
+
+def get_memory_size(ds: xr.Dataset, chunks: Dict[str, int]) -> int:
+    """Returns an estimate of memory size based on input chunks.
+    Currently this applies the chunks input to the dataset, then
+    iterates through the variables and returns the maximum.
+    """
+    ds_single_chunk = ds.isel({dim: slice(0, chunk) for dim, chunk in chunks.items()})
+    mem_size = max([ds_single_chunk[var].nbytes for var in ds_single_chunk.data_vars])
+    return mem_size
+
+
+def similarity(a: np.ndarray, b: np.ndarray) -> np.ndarray:
+    return np.sqrt(np.sum((a - b) ** 2))
+
+
+def normalize(a: np.ndarray) -> np.ndarray:
+    """Convert to a unit vector"""
+    return a / np.sqrt(np.sum(a**2))
+
+
+def even_divisor_chunks(n: int) -> List[int]:
+    """Returns values that evenly divide n"""
+    divisors = []
+    for i in range(1, n + 1):
+        if n % i == 0:
+            divisors.append(n // i)
+    return divisors
+
+
+# FIXME: mypy did not correctly infer the return type of parse_bytes
+def parse_bytes_wrapper(target_chunk_size: Union[str, int]) -> int:
+    return parse_bytes(target_chunk_size)
```

Comment:
If I do not wrap this with type hints, mypy is not able to detect that the result of parse_bytes is always an int and complains.

Reply:
Maybe @TomNicholas or @cisaacstern know more here. This confused me since the type hinting on `dask.utils.parse_bytes` seemed to be equivalent (see [here](https://github.com/dask/dask/blob/a8f2acc2e3328707cda2a25cee6cfadec44ec815/dask/utils.py#L1455))